src/crush/CrushCompiler.cc

```diff
@@ -673,23 +721,35 @@ int CrushCompiler::parse_rule(iter_t const& i)
 	  err << "in rule '" << rname << "' item '" << item << "' not defined" << std::endl;
 	  return -1;
 	}
+        int id = item_id[item];
         int c = -1;
         string class_name;
         if (s->children.size() > 2) {
           class_name = string_node(s->children[3]);
-          c = crush.rule_set_class(ruleno, step, class_name);
+          c = crush.get_class_id(class_name);
           if (c < 0)
             return c;
+          if (crush.class_bucket.count(id) == 0) {
+            err << "in rule '" << rname << "' step take " << item
+                << " has no class information" << std::endl;
+            return -EINVAL;
```

Comment:
this is a bug, right?  we should have ids for all classes and all buckets at this stage?